core/trino-main/src/main/java/io/trino/operator/RowChangeProcessor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.trino.spi.Page;
+import io.trino.spi.connector.RowChangeParadigm;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "@type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = DeleteAndInsertMergeProcessor.class, name = "deleteAndInsert"),
+        @JsonSubTypes.Type(value = ChangeOnlyUpdatedColumnsMergeProcessor.class, name = "changeOnlyUpdated"),
+})
+public interface RowChangeProcessor
+{
+    /**
+     * @return Return the RowChangeParadigm used by the RowChangeProcessor.
+     */
+    RowChangeParadigm getRowChangeParadigm();
+
+    /**
+     * Transform the inputPage into an output page.  The format of the inputPage depends
+     * on the operation.  For SQL MERGE, the inputPage consists of the write redistribution
+     * column values; followed by the rowId block; followed by the merge case RowBlock;
+     * followed by a hash column block for partitioned or bucketed tables.
+     * The result is a page starting with all non-hidden columns computed by the SQL MERGE
+     * operation, in table column order, followed by rowId column, followed by the "operation"
+     * block whose values are
+     * @param inputPage For SQL MERGE, the inputPage consists of the write redistribution
+     * columns, if any, followed by the rowId block for the row in the target table,
+     * followed by merge case RowBlock, followed by a hash column block for partitioned
+     * or bucketed tables.
```

Comment:
Removed.  Looking again at the type hierarchy, I inserted new interface `MergeRowChangeProcessor` between `RowChangeProcessor` and both `DeleteAndInsertMergeProcessor` and `ChangeOnlyUpdatedColumnsMergeProcessor`.  This makes sense because the inputPage is always the same for all `MERGE` implementors of `RowChangeProcessor`, and the output page is nearly the same among implementors.